### PR TITLE
Add EIP-6464: Multi-operator, per-token ERC721 approval

### PR DESCRIPTION
**Pull request by aschlosberg**
### Abstract

> The ERC721 standard did not foresee the approval of multiple operators to manage a specific token on behalf of its owner. This lead to the establishment of `setApprovalForAll()` as the predominant way to authorise operators, which affords the approved address control over all assets and creates an unnecessarily broad security risk that has already been exploited in a multitude of phishing attacks. The presented EIP extends ERC721 by introducing a fine-grained, on-chain approval mechanism that allows owners to authorise multiple, specific operators on a per-token basis; this removes unnecessary access permissions and shrinks the surface for exploits to a minimum. The provided reference implementation further enables cheap revocation of all approvals on a per-owner or per-token basis.

### TL;DR

Extend ERC721 with `setExplicitApproval(address operator, uint256 tokenId, bool approved)` to allow NFT owners to avoid use of `setApprovalForAll()` when selling on multiple marketplaces.

### CODE REVIEW

**Comment by aschlosberg** [comment]
We've addressed all linter issues, including `eth-bot`'s (using the presumptive EIP number 6464 based on the PR; it's a nice number so we'd love to keep it if possible, please).

I'll try get `eth-bot` to rerun in another comment but am guessing permissions won't allow it :)

**Comment by Pandapip1** [comment]
@eth-bot rerun

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please rebase the branch. Your walidator version is outdated.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please fix [the walidator issues](https://github.com/ethereum/EIPs/pull/6464/files)

**Comment by aschlosberg** [comment]
@eth-bot rerun

**Comment by aschlosberg** [comment]
Thanks for taking a look, @Pandapip1, all walidator issues have been addressed.

File: assets/eip-6464/contracts/IERC165.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-6464/contracts/IERC165.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-6464/contracts/IERC6464.sol [hunk 2]
```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-6464/contracts/IERC6464.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-6464/contracts/IERC721.sol [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-6464/contracts/IERC721.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by aschlosberg** [comment]
I couldn't just swap out MIT for CC0 as those were OpenZeppelin versions, so I've replaced them with the originals from the EIPs themselves. Also changed ours to CC0 as requested.

File: EIPS/eip-6464.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+1. Security measures for protecting NFTs other than through limiting the scope of operator approvals.
+2. Compatibility with [ERC-1155](./eip-1155.md) semi-fungible tokens. However we note that the mechanisms described herein are also applicable to ERC-1155 token *types* without requiring approval for all other types.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+To comply with this EIP, a contract MUST implement `IERC6464` (defined herein) and the `ERC165` and `ERC721` interfaces; see [ERC-165](./eip-165.md) and ERC-721 respectively.
+
+```
```
Comment (Pandapip1) [inline EIPS/eip-6464.md:46]:
```suggestion
```solidity
```

**Review by Pandapip1 — APPROVED** [review]
I'm still not sure that I understand this, but it looks complete and has no obvious errors. Hopefully you can make this a bit clearer before Review.

**Comment by aschlosberg** [comment]
> I'm still not sure that I understand this, but it looks complete and has no obvious errors. Hopefully you can make this a bit clearer before Review.

Thanks for the reviews, @Pandapip1. Are we allowed to include lay descriptions in the Abstract? The template says it must be technical and very terse, but I think something like the following would go a long way to clarify:

> The only way to sell NFTs on more than one marketplace simultaneously is to approve each marketplace to manage _all_ of your tokens on a contract, not just the single token you wish to sell. This EIP introduces a fine-grained way to approve only specific token(s) and thus reduce the security risk from phishing scams that piggyback off marketplace contracts.

**Pull request merged**